api/src/routes/donate.ts

```diff
@@ -76,5 +87,141 @@ export const donateRoutes: FastifyPluginCallbackTypebox = (
     }
   );
 
+  fastify.post(
+    '/donate/charge-stripe-card',
+    {
+      schema: schemas.chargeStripeCard
+    },
+    async (req, reply) => {
+      try {
+        const { paymentMethodId, amount, duration } = req.body;
+        const { id } = req.session.user;
+
+        const user = await fastify.prisma.user.findUnique({
+          where: { id }
+        });
+
+        if (!user) {
+          void reply.code(500);
+          return {
+            type: 'danger',
+            message: 'Something went wrong.'
+          } as const;
+        }
+
+        const { email, name } = user;
+
+        if (user.isDonating) {
```

Comment:
```suggestion
        if (user.isDonating && duration !== 'one-time') {
```
That's the current logic.

Reply:
Yes, but we don't allow card payments to donors regardless of the donation duration.